urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/resource/DefaultDirectoryResource.java

```diff
@@ -0,0 +1,533 @@
+		lineBuffer.delete( 0, lineBuffer.length() );
+		
+	    }
+
+	    return -1;
+
+	} else {
+
+	    List<String[]> tableData = new ArrayList<String[]>( files.length );
+	    int[] rowSizes = null;
+	    String[] rowData = null;
+	    for( int i = 0; i < files.length; i++ ) {
+		
+		rowData = collectRowData( files[i] );
+		tableData.add( rowData );
+
+		if( rowSizes == null )
+		    rowSizes = new int[ rowData.length ];
+
+		for( int c = 0; c < rowData.length; c++ ) {
+
+		    rowSizes[c] = Math.max( rowSizes[c], rowData[c].length() );
+		}
+
+	    } // END for
+	    
+	    // Generate text output
+	    int totalWidth = 80;     // 80 for the case there are not files
+	    for( int i = 0; i < tableData.size(); i++ ) {		
+
+		rowData = tableData.get( i );
+		buildTextLine( lineBuffer,
+			       rowData,
+			       rowSizes,
+			       4    // columnSpacing
+			       );
+		out.write( lineBuffer.toString().getBytes(java.nio.charset.StandardCharsets.UTF_8.name()) );
+		totalWidth = lineBuffer.length() - 1;  // do not include line break
+		lineBuffer.delete( 0, lineBuffer.length() );
+
+	    }
+
+	    return totalWidth;
+	}
+    }
+
+    private String[] collectRowData( File file ) {
+	// 0: icon
+	// 1: name
+	// 2: size
+	// 3: type
+	// 4: date
+	String[] row = new String[ 5 ];
+	
+	for( int i = 0; i < row.length; i++ ) {
+	    row[i] =  getFileAttribute( file, i );
+	}
+
+	return row;
+    }
+
+    private void buildTextLine( StringBuffer lineBuffer,
+				String[] rowData,
+				int[] rowSizes,
+				int columnSpacing) {
+
+	for( int i = 0; i < rowData.length; i++ ) {
+
+	    String data = rowData[i];
+	    int len     = rowSizes[i];
+	    
+	    if( i == 2 ) {
+
+		// The length should be aligned right
+		lineBuffer.append( CustomUtil.repeat( " ", len - data.length()) );
+		lineBuffer.append( data );
+		
+	    } else {
+
+		// Other values should be aligned left	
+		lineBuffer.append( data );
+		lineBuffer.append( CustomUtil.repeat( " ", len - data.length()) );
+
+	    }
+
+	    // Print column spacing
+	    lineBuffer.append( CustomUtil.repeat( " ", columnSpacing) );
+	    
+	}
+
+	lineBuffer.append( "\n" );
+
+    }
+
+    private void buildHTMLLine( StringBuffer lineBuffer,
+				File file ) {
+	
+	if( this.isHTMLFormat() ) 
+	    lineBuffer.append( "<tr>\n" );
+
+	    
+	for( int i = 0; i < 5; i++ ) {
+	    
+	    if( this.isHTMLFormat() ) {
+		lineBuffer.append( "      <td" );
+		if( i == 0 || i == 2 ) // icon or size
+		    lineBuffer.append( " align=\"right\"" );
+		lineBuffer.append( ">" );
+	    } else if( i > 0 ) {
+		lineBuffer.append( "\t\t" );
+	    }
+	    
+
+	    lineBuffer.append( getFileAttribute(file, i) );
+
+
+	    if( this.isHTMLFormat() )
+		lineBuffer.append( "</td>\n" );
+	    
+	}
+
+
+	if( this.isHTMLFormat() )
+	    lineBuffer.append( "</tr>\n\n" );
+	else
+	    lineBuffer.append( "\n" );
+
+    }
+
+    private String getFileAttribute( File file,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.